src/tpm_socket_comm.c

```diff
@@ -57,17 +58,19 @@ static int add_to_buffer(TPM_SOCKET_INFO* socket_info, const unsigned char* byte
     }
     else
     {
-        new_buff = (unsigned char*)realloc(socket_info->recv_bytes, socket_info->recv_length + length);
-}
+        size_t realloc_size = safe_add_size_t(socket_info->recv_length, length);
+        new_buff = (unsigned char*)realloc(socket_info->recv_bytes, realloc_size);
+    }
+
     if (new_buff == NULL)
     {
         result = MU_FAILURE;
     }
     else
     {
         socket_info->recv_bytes = new_buff;
-        memcpy(socket_info->recv_bytes + socket_info->recv_length, bytes, length);
-        socket_info->recv_length += length;
+        memcpy(safe_add_size_t(socket_info->recv_bytes, socket_info->recv_length), bytes, length);
```

Comment:
same here, does the return value of safe_add_size_t need to be checked?

Reply:
reverted